### PR TITLE
nixpkgs/NixOS manuals: devmode feature

### PR DESCRIPTION
**Pull request by mightyiam**
This does not add anything to the NixOS manual intentionally.
I intend to tackle that in #241871.

Co-authored-by: Alejandro Sanchez Medina <alejandrosanchzmedina@gmail.com>

### CODE REVIEW

File: doc/default.nix [hunk 1]
```diff
@@ -85,11 +85,22 @@ let
...
+    inherit outputPath;
+    indexPath = "/manual.html";
+  };
+
 in pkgs.stdenv.mkDerivation {
   name = "nixpkgs-manual";
 
   nativeBuildInputs = with pkgs; [
     nixos-render-docs
+    web-devmode
```
Comment (jtojnar) [inline doc/default.nix:103]: What is the point of adding it here?

**Review by pennae — CHANGES_REQUESTED** [review]

File: doc/default.nix [hunk 2]
```diff
@@ -85,11 +85,22 @@ let
...
         };
   };
+
+  outputPath="share/doc/nixpkgs";
+
+  web-devmode = import ../pkgs/tools/nix/web-devmode.nix {
+    inherit pkgs;
+    buildCommand = "nix-build --out-link $out_link";
+    inherit outputPath;
+    indexPath = "/manual.html";
```
[the remark this answers is not in the document]
Comment (mightyiam) [reply doc/default.nix:95]: I'm not sure what kind of problems you're thinking about. To be clear, this is a URL path. This is the path that will be opened in the browser. It is passed, as is, as an argument to `live-server` CLI. So, I suppose it should be absolute.

File: pkgs/tools/nix/web-devmode.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  # A Rust alternative to live-server exists, but it was not in nixpkgs.
+  # `--no-css-inject`: without this flag, it seems that only CSS is auto-reloaded.
+  # https://www.npmjs.com/package/live-server
+  server = pkgs.writeShellScriptBin "server" ''
+    ${pkgs.nodePackages_latest.live-server + /bin/live-server} \
+      --verbose \
+      --no-css-inject \
+      --open=${indexPath} \
+      $serve
```
[the remark this answers is not in the document]
Comment (mightyiam) [reply pkgs/tools/nix/web-devmode.nix]: Haven't been able to resolve this. Perhaps with something other than `live-server`... Not worth it, I feel.

**Review by pennae — COMMENTED** [review]
this is getting somewhere :) mostly nitpicks left now, overall it's looking pretty good.

**Review by pennae — APPROVED** [review]
there's still a corner case of the live server getting a bit stuck when you rename the output file you're currently viewing, in that case it'll just show a "could not GET" message and won't reload anything. but that's almost certainly a `live-server` issue, so we'd call this good now.

**Pull request merged**